### PR TITLE
feat: add Subcommunities page

### PR DESCRIPTION
**Pull request by carlinmack**
:heart: Thank you for your contribution!

Close https://github.com/zenodo/zenodo-rdm/issues/990

### Description

> Please describe briefly your pull request.
image

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [ ] I've added relevant documentation.
- [ ] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [ ] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [ ] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [ ] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [ ] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [ ] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: invenio_communities/communities/services/config.py [hunk 1]
```diff
@@ -114,6 +114,10 @@ class CommunityServiceConfig(RecordServiceConfig, ConfiguratorMixin):
         "invitations": CommunityLink("{+api}/communities/{id}/invitations"),
         "requests": CommunityLink("{+api}/communities/{id}/requests"),
         "records": CommunityLink("{+api}/communities/{id}/records"),
+        "subcommunities": CommunityLink(
+            "{+api}/communities/{id}/subcommunities",
```
Comment (carlinmack) [inline invenio_communities/communities/services/config.py:118]:
```suggestion
            "{+api}/communities/{id}/browse/subcommunities",
```

File: invenio_communities/subcommunities/resources/config.py [hunk 2]
```diff
@@ -30,6 +37,7 @@ class SubCommunityResourceConfig(ConfiguratorMixin, ResourceConfig):
     url_prefix = ""
     routes = {
         "join": "/communities/<pid_value>/actions/join-request",
+        "list": "/communities/<pid_value>/subcommunities",
```
Comment (carlinmack) [inline invenio_communities/subcommunities/resources/config.py:40]:
```suggestion
        "list": "/communities/<pid_value>/browse/subcommunities",
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/subcommunity/search.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of Invenio.
+ * Copyright (C) 2016-2021 CERN.
```
Comment (alejandromumo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/subcommunity/search.js:3]:
```suggestion
 * Copyright (C) 2024 CERN.
```

File: invenio_communities/templates/semantic-ui/invenio_communities/details/subcommunity/index.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+{%- block page_body %}
+{{ super() }}
+<div class="ui container fluid rel-mt-3 rel-mb-2 compact ml-0-mobile mr-0-mobile">
+  <div class="ui container community-search page-subheader flex align-items-center justify-space-between">
+    <h1 class="ui large header m-0">{{_("Subcommunities")}}</h1>
+  </div>
+</div>
+<div id="communities-search" class="rel-mt-2"
+    data-invenio-search-config='{{search_app_communities_subcommunities_config(app_id="InvenioCommunities.Search", endpoint=search_endpoint) | tojson }}'
```
Comment (alejandromumo) [inline invenio_communities/templates/semantic-ui/invenio_communities/details/subcommunity/index.html:29]:
```suggestion
    data-invenio-search-config='{{search_app_communities_subcommunities_config(app_id="InvenioSubCommunities.Search", endpoint=search_endpoint) | tojson }}'
```

File: invenio_communities/assets/semantic-ui/js/invenio_communities/subcommunity/search.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+import { overrideStore, parametrize } from "react-overridable";
+import {
+  CommunitiesResults,
+  CommunitiesSearchBarElement,
+  CommunitiesSearchLayout,
+  CommunityItem,
+  ResultsGridItemTemplate,
+} from "../community";
+
+const appName = "InvenioCommunities.Search";
```
Comment (alejandromumo) [inline invenio_communities/assets/semantic-ui/js/invenio_communities/subcommunity/search.js:24]:
```suggestion
const appName = "InvenioSubCommunities.Search";
```

File: invenio_communities/searchapp.py [hunk 6]
```diff
@@ -25,6 +25,14 @@ def search_app_context():
             pagination_options=(10, 20),
             endpoint="/api/communities",
         ),
+        "search_app_communities_subcommunities_config": partial(
+            search_app_config,
+            config_name="COMMUNITIES_SUBCOMMUNITIES_SEARCH",
+            available_facets=current_app.config["COMMUNITIES_SUBCOMMUNITIES_FACETS"],
+            sort_options=current_app.config["COMMUNITIES_SORT_OPTIONS"],
+            headers={"Accept": "application/vnd.inveniordm.v1+json"},
+            pagination_options=(10, 20),
```
Comment (alejandromumo) [inline invenio_communities/searchapp.py:34]: 😛 

```suggestion
            pagination_options=(10, 25, 50, 100),
```

File: invenio_communities/config.py [hunk 7]
```diff
@@ -26,6 +26,7 @@
     "upload": "/communities/<pid_value>/upload",
     "settings": "/communities/<pid_value>/settings",
     "requests": "/communities/<pid_value>/requests",
+    "subcommunities": "/communities/<pid_value>/browse/subcommunities",
```
Comment (alejandromumo) [inline invenio_communities/config.py:29]: Shouldn't this be as defined below by the resource? 

e.g. 
`/communities/<pid_value>/subcommunities`

The concept of `browse` does not exist in `invenio-communities` at this point

File: invenio_communities/subcommunities/resources/resource.py [hunk 8]
```diff
@@ -42,3 +46,18 @@ def join(self):
             data=resource_requestctx.data,
         )
         return result.to_dict(), 201
+
+    @request_view_args
+    @response_handler(many=True)
+    @request_extra_args
+    @request_search_args
+    def search(self):
```
Comment (alejandromumo) [inline invenio_communities/subcommunities/resources/resource.py:54]: nit: shouldn't this be defined by the `communities` resource instead? E.g. I want to search the subcommunities of a community.

File: invenio_communities/templates/semantic-ui/invenio_communities/details/subcommunity/index.html [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+    This file is part of Invenio.
+    Copyright (C) 2024 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+#}
+
+{% extends "invenio_communities/details/base.html" %}
+
+{% set active_community_header_menu_item= 'browse' %}
```
Comment (alejandromumo) [inline invenio_communities/templates/semantic-ui/invenio_communities/details/subcommunity/index.html:12]: Perhaps we could define a macro to render subcommunities here and use it in `invenio-app-rdm` in the `browse` page?

**Review by alejandromumo — APPROVED** [review]

**Pull request merged**